### PR TITLE
ci(deploy-updates): trigger rumba update on new release (MP-222)

### PR DESCRIPTION
**Pull request by Guyzeroth**

### CODE REVIEW

**Review by caugner — CHANGES_REQUESTED** [review]
To avoid duplication, could you put the commands in a bash script and use env vars to parameterise ?

File: .github/workflows/deploy-prod-updates.yml [hunk 1]
```diff
@@ -34,3 +34,20 @@ jobs:
           aws-secret-access-key: ${{ secrets.UPDATES_PROD_AWS_SECRET_ACCESS_KEY }}
           aws-region: us-west-2
       - run: aws s3 sync --acl public-read rumba-updates/ s3://updates-prod-developer-mozilla-e2b95ed55d379b1a/rumba-bcd-updates/
```
Comment (LeoMcA) [inline .github/workflows/deploy-prod-updates.yml:36]: Do we need to invalidate CDN after this step, or doesn't this file get cached?

File: .github/workflows/deploy-prod-updates.yml [hunk 2]
```diff
@@ -34,3 +34,20 @@ jobs:
...
+    env:
+      RUMBA_AUTH: ${{ secrets.RUMBA_PROD_API_KEY }}
+    steps:
+      - run: |
+          most_recent_update=`curl https://updates.developer.mozilla.org/rumba-bcd-updates/bcd-updates.json | jq '[.browsers[].releases[] | select(.status == "current")] | max_by(.release_date) | .release_date'`
+          most_recent_api_release=`curl https://developer.mozilla.org/api/v2/updates/ | jq '.data[0].release_date'`
+          if [ $(date -d $most_recent_update +%s) -gt $(date -d $most_recent_api_release +%s) ]; 
+          then 
+            echo "$most_recent_update is more recent than $most_recent_api_release";
+            curl -X POST 'https://developer.mozilla.org/admin-api/v2/updates/' --header "Authorization: Bearer $RUMBA_AUTH"
```
Comment (LeoMcA) [inline .github/workflows/deploy-prod-updates.yml:50]: can we do `curl --fail` to get it to fail the job if this returns an internal server error?

File: .github/workflows/deploy-prod-updates.yml (hunk 1 above)
Comment (Guyzeroth) [reply .github/workflows/deploy-prod-updates.yml:36]: This file does not. But you have reminded me that on successful update from API we do need to invalidate the /api/v2/updates/ cache.

File: scripts/sync_rumba_updates.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+most_recent_update=`curl --fail $updates_host/rumba-bcd-updates/bcd-updates.json | jq '[.browsers[].releases[] | select(.status == "current")] | max_by(.release_date) | .release_date'`
+most_recent_api_release=`curl --fail $api_host/api/v2/updates/ | jq '.data[0].release_date'`
+if [ $(date -d $most_recent_update +%s) -gt $(date -d $most_recent_api_release +%s) ]; 
+    then 
+        echo "$most_recent_update is more recent than $most_recent_api_release";
+        curl --fail -X POST $api_host/admin-api/v2/updates/ --header "Authorization: Bearer $RUMBA_AUTH"
+    else
+        echo "$most_recent_update is not more recent than $most_recent_api_release. Finished!";
+fi
```
Comment (caugner) [inline scripts/sync_rumba_updates.sh:12]:
```suggestion
fi

```

File: .github/workflows/deploy-prod-updates.yml [hunk 4]
```diff
@@ -34,3 +34,24 @@ jobs:
           aws-secret-access-key: ${{ secrets.UPDATES_PROD_AWS_SECRET_ACCESS_KEY }}
           aws-region: us-west-2
       - run: aws s3 sync --acl public-read rumba-updates/ s3://updates-prod-developer-mozilla-e2b95ed55d379b1a/rumba-bcd-updates/
+  trigger-prod-rumba-update-if-new:
```
Comment (caugner) [inline .github/workflows/deploy-prod-updates.yml:37]:
```suggestion

  trigger-prod-rumba-update-if-new:
```

File: .github/workflows/deploy-stage-updates.yml [hunk 5]
```diff
@@ -34,3 +34,24 @@ jobs:
           aws-secret-access-key: ${{ secrets.UPDATES_STAGE_AWS_SECRET_ACCESS_KEY }}
           aws-region: us-west-2
       - run: aws s3 sync --acl public-read rumba-updates/ s3://updates-stage-developer-allizom-6d533edfe2c2c683/rumba-bcd-updates/
+  trigger-stage-rumba-update-if-new:
```
Comment (caugner) [inline .github/workflows/deploy-stage-updates.yml:37]:
```suggestion

  trigger-stage-rumba-update-if-new:
```

File: .github/workflows/deploy-stage-updates.yml [hunk 6]
```diff
@@ -34,3 +34,24 @@ jobs:
...
+    name: Trigger rumba updates if new release.
+    needs: deploy-stage-updates
+    runs-on: ubuntu-latest
+    env:
+      RUMBA_AUTH: ${{ secrets.RUMBA_STAGE_API_KEY }}
+    steps:
```
Comment (caugner) [inline .github/workflows/deploy-stage-updates.yml:43]:
```suggestion

    steps:
```

**Review by caugner — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-prod-updates.yml [hunk 7]
```diff
@@ -34,3 +34,29 @@ jobs:
...
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          aws-access-key-id: ${{ secrets.PROD_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.PROD_AWS_ACCESS_KEY_SECRET }}
+          aws-region: us-west-2
+
+      - name: Invalidate Prod API Updates CDN
+        env:
+          DISTRIBUTION: E2ZY2DGUN70EMI
+          PATHS: api/v2/updates/
```
Comment (caugner) [inline .github/workflows/deploy-prod-updates.yml:61]:
```suggestion
          PATHS: api/v2/updates/*
```

File: .github/workflows/deploy-stage-updates.yml [hunk 8]
```diff
@@ -34,3 +34,31 @@ jobs:
...
+        uses: aws-actions/configure-aws-credentials@v1-node16
+        with:
+          aws-access-key-id: ${{ secrets.STAGE_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.STAGE_AWS_ACCESS_KEY_SECRET }}
+          aws-region: us-west-2
+
+      - name: Invalidate Stage API Updates CDN
+        env:
+          DISTRIBUTION: E2MLRMA1VTVDHX
+          PATHS: api/v2/updates/
```
Comment (caugner) [inline .github/workflows/deploy-stage-updates.yml:63]:
```suggestion
          PATHS: api/v2/updates/*
```

**Review by caugner — CHANGES_REQUESTED** [review]
It seems invalidation paths seem to have a wildcard, see: https://github.com/mdn/bcd-utils/actions/runs/4317213188/jobs/7534007011#step:5:11

File: scripts/sync_rumba_updates.sh [hunk 9]
```diff
@@ -0,0 +1,12 @@
+updates_host=${UPDATES_HOST?};
+api_host=${RUMBA_HOST?};
+
+most_recent_update=`curl --fail $updates_host/rumba-bcd-updates/bcd-updates.json | jq '[.browsers[].releases[] | select(.status == "current")] | max_by(.release_date) | .release_date'`
+most_recent_api_release=`curl --fail $api_host/api/v2/updates/ | jq '.data[0].release_date'`
+if [ $(date -d $most_recent_update +%s) -gt $(date -d $most_recent_api_release +%s) ];
```
Comment (caugner) [inline scripts/sync_rumba_updates.sh:6]: Looks like this fails, see: https://github.com/mdn/bcd-utils/actions/runs/4317279119/jobs/7534154507#step:3:21

```
...
date: invalid date ‘"2023-02-14"’
date: invalid date ‘"2023-02-14"’
"2023-02-14" is more recent than "2023-02-14"
```

**Review by caugner — CHANGES_REQUESTED** [review]
Looks good, but doesn't seem to work. 😕

File: scripts/sync_rumba_updates.sh (hunk 9 above)
Comment (caugner) [reply scripts/sync_rumba_updates.sh:6]: And `jq -r` should do the trick, see: https://stackoverflow.com/a/44656583

Comment (caugner) [reply scripts/sync_rumba_updates.sh:6]: Fixed via f4f8617589623902278a150817f334d0abf81d3c.

Now it looks like expected: `2023-02-14 is not more recent than 2023-02-14. Finished!`

https://github.com/mdn/bcd-utils/actions/runs/4317320441/jobs/7534254246#step:3:20

File: .github/workflows/deploy-prod-updates.yml (hunk 2 above)
Comment (caugner) [reply .github/workflows/deploy-prod-updates.yml]: Resolved via a16267808705afd93514bc6cdd218bf594ff55d8.

File: .github/workflows/deploy-prod-updates.yml (hunk 1 above)
Comment (caugner) [reply .github/workflows/deploy-prod-updates.yml:36]: Resolved via a16267808705afd93514bc6cdd218bf594ff55d8.

**Review by caugner — APPROVED** [review]

**Review by caugner — COMMENTED** [review]
Let's wait with merging until there is a new (stable) browser release, so that we can consciously test the workflow once more.

**Review by caugner — CHANGES_REQUESTED** [review]
It doesn't work. The workflows run successfully, but the Updates data stays outdated. Subsequent runs also determine that the data is still outdated.

**Review by caugner — APPROVED** [review]
✅ All good, works on [stage](https://github.com/mdn/bcd-utils/actions/runs/4420473253) and [prod](https://github.com/mdn/bcd-utils/actions/runs/4420495392).

**Pull request merged**